yuzu-mainline doesn't build on 21.05

**Describe the bug**
A clear and concise description of what the bug is.

`yuzu-mainline` fails to build.

**To Reproduce**
Steps to reproduce the behavior:
1. add `yuzu-mainline` to `environment.systemPackages`.
2. `nixos-rebuild switch`
3. observe failure

**Expected behavior**
A clear and concise description of what you expected to happen.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
Add any other context about the problem here.

most relevant part of build log:
```
[ERROR] fatal: git cat-file: could not get object info
[ERROR] Unrecognized git object type:
[ERROR] Unable to checkout eac2416b8fdb2cb9c867a538100bf95326bad75b from https://github.com/yuzu-emu/mbedtls.
[ERROR] builder for '/nix/store/dqa14b10hb9w2s4758v6zqhr9wyn9m5g-source.drv' failed with exit code 1
```

**Notify maintainers**
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
@IvarWithoutBones 

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```
builtins.fetchTarball {
  name = "nixos-stable-2021-06-25";
  url = "https://github.com/nixos/nixpkgs/archive/b72bde7c4a1f9c9bf1a161f0c267186ce3c6483c.tar.gz";
  sha256 = "1cihs542msfldcawcsk6s2bs8bpz82ldy6axsrax8ngw2gllqrcd";
}
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: yuzu-mainline
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/misc/emulators/ryujinx/default.nix
+++ b/pkgs/misc/emulators/ryujinx/default.nix
@@ -1,7 +1,7 @@
 { lib, stdenv, fetchFromGitHub, fetchurl, makeWrapper, makeDesktopItem, linkFarmFromDrvs
 , dotnet-sdk_5, dotnetPackages, dotnetCorePackages, cacert
 , libX11, libgdiplus, ffmpeg
-, SDL2_mixer, openal, libsoundio, sndio
+, SDL2_mixer, openal, libsoundio, sndio, pulseaudio
 , gtk3, gobject-introspection, gdk-pixbuf, wrapGAppsHook
 }:
 
@@ -15,16 +15,17 @@ let
     openal
     libsoundio
     sndio
+    pulseaudio
   ];
 in stdenv.mkDerivation rec {
   pname = "ryujinx";
-  version = "1.0.6893"; # Versioning is based off of the official appveyor builds: https://ci.appveyor.com/project/gdkchan/ryujinx
+  version = "1.0.6954"; # Versioning is based off of the official appveyor builds: https://ci.appveyor.com/project/gdkchan/ryujinx
 
   src = fetchFromGitHub {
     owner = "Ryujinx";
     repo = "Ryujinx";
-    rev = "fb65f392d1c4b0e01f22b6ddebcc8317ba9769c3";
-    sha256 = "0ncrcbwyirz21j81vc6yvcjc2fn8nc5ilrfczclq6f8cpy09c3y8";
+    rev = "31cbd09a75a9d5f4814c3907a060e0961eb2bb15";
+    sha256 = "00qql0wmlzs722s0igip3v0yjlqhc31jcr7nghwibcqrmx031azk";
   };
 
   nativeBuildInputs = [ dotnet-sdk_5 dotnetPackages.Nuget cacert makeWrapper wrapGAppsHook gobject-introspection gdk-pixbuf ];
@@ -79,9 +80,13 @@ in stdenv.mkDerivation rec {
       --output $out/lib/ryujinx
     shopt -s extglob
 
+    # TODO: fix this hack https://github.com/Ryujinx/Ryujinx/issues/2349
+    mkdir -p $out/lib/sndio-6
+    ln -s ${sndio}/lib/libsndio.so $out/lib/sndio-6/libsndio.so.6
+
     makeWrapper $out/lib/ryujinx/Ryujinx $out/bin/Ryujinx \
       --set DOTNET_ROOT "${dotnetCorePackages.net_5_0}" \
-      --suffix LD_LIBRARY_PATH : "${lib.makeLibraryPath runtimeDeps}" \
+      --suffix LD_LIBRARY_PATH : "${builtins.concatStringsSep ":" [ (lib.makeLibraryPath runtimeDeps) "$out/lib/sndio-6" ]}" \
       ''${gappsWrapperArgs[@]}
 
     for i in 16 32 48 64 96 128 256 512 1024; do

--- a/pkgs/misc/emulators/ryujinx/deps.nix
+++ b/pkgs/misc/emulators/ryujinx/deps.nix
@@ -10,6 +10,11 @@
     sha256 = "1rjdxd4fq5z3n51qx8vrcaf4i277ccc62jxk88xzbsxapdmjjdf9";
   })
   (fetchNuGet {
+    name = "CommandLineParser";
+    version = "2.8.0";
+    sha256 = "1m32xyilv2b7k55jy8ddg08c20glbcj2yi545kxs9hj2ahanhrbb";
+  })
+  (fetchNuGet {
     name = "Concentus";
     version = "1.1.7";
     sha256 = "0y5z444wrbhlmsqpy2sxmajl1fbf74843lvgj3y6vz260dn2q0l0";
@@ -260,11 +265,6 @@
     sha256 = "0dkl9j0yd65s5ds9xj5z6yb7yca7wlycqz25m8dng20d13sqr1zp";
   })
   (fetchNuGet {
-    name = "ppy.SDL2-CS";
-    version = "1.0.225-alpha";
-    sha256 = "1x8hvk9kikwi7wrvwxdxk3pkbs491iss2mvqgiw844zld5izihqc";
-  })
-  (fetchNuGet {
     name = "runtime.any.System.Collections";
     version = "4.3.0";
     sha256 = "0bv5qgm6vr47ynxqbnkc7i797fdi8gbjjxii173syrx14nmrkwg0";
@@ -513,6 +513,11 @@
     name = "Ryujinx.Graphics.Nvdec.Dependencies";
     version = "4.4.0-build7";
     sha256 = "0g1l3lgs0ffxp64ka81v6q1cgsdirl1qlf73255v29r3v337074m";
+  })
+  (fetchNuGet {
+    name = "Ryujinx.SDL2-CS";
+    version = "2.0.15-build11";
+    sha256 = "0s4h69l2b508l5wxp4v4ip8k83k78p3963xxv8bfamin9517przi";
   })
   (fetchNuGet {
     name = "SharpZipLib";

--- a/pkgs/misc/emulators/yuzu/base.nix
+++ b/pkgs/misc/emulators/yuzu/base.nix
@@ -1,5 +1,5 @@
 { pname, version, src, branchName
-, stdenv, lib, fetchFromGitHub, fetchpatch, wrapQtAppsHook
+, stdenv, lib, wrapQtAppsHook
 , cmake, pkg-config
 , libpulseaudio, libjack2, alsa-lib, sndio
 , vulkan-loader, vulkan-headers
@@ -9,7 +9,7 @@
 , glslang
 , boost173
 , catch2
-, fmt
+, fmt_8
 , SDL2
 , udev
 , libusb1
@@ -29,18 +29,11 @@ stdenv.mkDerivation rec {
     glslang
     boost173
     catch2
-    fmt
+    fmt_8
     SDL2
     udev
     libusb1
     ffmpeg
-  ];
-
-  patches = [
-    (fetchpatch { # Without this, yuzu tries to read version info from .git which is not present.
-      url = "https://raw.githubusercontent.com/pineappleEA/Pineapple-Linux/28cbf656e3188b80eda0031d0b2713708ecd630f/inject-git-info.patch";
-      sha256 = "1zxh5fwdr7jl0aagb3yfwd0995vyyk54f0f748f7c4rqvg6867fd";
-    })
   ];
 
   cmakeFlags = [
@@ -50,10 +43,11 @@ stdenv.mkDerivation rec {
     "-DENABLE_QT_TRANSLATION=ON"
     "-DYUZU_USE_QT_WEB_ENGINE=ON"
     "-DUSE_DISCORD_PRESENCE=ON"
-    # Shows errors about not being able to find .git at runtime if you do not set these
-    "-DGIT_BRANCH=\"\""
-    "-DGIT_DESC=\"\""
   ];
+
+  # This changes `ir/opt` to `ir/var/empty` in `externals/dynarmic/src/dynarmic/CMakeLists.txt`
+  # making the build fail, as that path does not exist
+  dontFixCmake = true;
 
   preConfigure = ''
     # Trick the configure system. This prevents a check for submodule directories.
@@ -61,8 +55,8 @@ stdenv.mkDerivation rec {
 
     # see https://github.com/NixOS/nixpkgs/issues/114044, setting this through cmakeFlags does not work.
     cmakeFlagsArray+=(
-      "-DTITLE_BAR_FORMAT_IDLE=\"yuzu ${branchName} ${version}\""
-      "-DTITLE_BAR_FORMAT_RUNNING=\"yuzu ${branchName} ${version} \| \{3\}\""
+      "-DTITLE_BAR_FORMAT_IDLE=yuzu ${branchName} ${version}"
+      "-DTITLE_BAR_FORMAT_RUNNING=yuzu ${branchName} ${version} | {3}"
     )
   '';
 
@@ -87,5 +81,6 @@ stdenv.mkDerivation rec {
     ];
     maintainers = with maintainers; [ ivar joshuafern ];
     platforms = platforms.linux;
+    broken = stdenv.isAarch64; # Currently aarch64 is not supported.
   };
 }

--- a/pkgs/misc/emulators/yuzu/default.nix
+++ b/pkgs/misc/emulators/yuzu/default.nix
@@ -4,25 +4,25 @@ let
 in {
   mainline = libsForQt5.callPackage ./base.nix rec {
     pname = "yuzu-mainline";
-    version = "633";
+    version = "679";
     branchName = branch;
     src = fetchFromGitHub {
       owner = "yuzu-emu";
       repo = "yuzu-mainline";
       rev = "mainline-0-${version}";
-      sha256 = "1zq20dd6x3kk179ls7y3mc5rj9vr73qs12bdj52bl3kscphxybf1";
+      sha256 = "1wkxkgfff5nn30mn2pqcrlsr7fja6bgaallsx1ainlcbamfyspcz";
       fetchSubmodules = true;
     };
   };
   early-access = libsForQt5.callPackage ./base.nix rec {
     pname = "yuzu-ea";
-    version = "1704";
+    version = "1855";
     branchName = branch;
     src = fetchFromGitHub {
       owner = "pineappleEA";
       repo = "pineapple-src";
       rev = "EA-${version}";
-      sha256 = "07pcldcmn8ammzbnxki6by7hgacr6z0548dvhv2p5lp9qcf3xz22";
+      sha256 = "0civ63zi23cym3kii4v3aiqrsg2zzj0w6sp6sa14ixysagqxsd4a";
     };
   };
 }.${branch}

